UndertaleModTool/Scripts/Builtin Scripts/RunSwitchAndXboxOnPC.csx

```diff
@@ -0,0 +1,172 @@
+ReplaceTextInGML("gml_Script_control_update", "global.osflavor >= 4", "obj_time.j_ch > 0");
+
+// Fix Joystick Menu
+ReplaceTextInGML("gml_Object_obj_joypadmenu_Create_0", "joystick_has_pov(obj_time.j_ch)", "true");
+ReplaceTextInGML("gml_Object_obj_joypadmenu_Draw_0", "joystick_has_pov(obj_time.j_ch)", "true");
+// gamepad_button_count(obj_time.j_ch - 1) might work better but I'm not sure
+ReplaceTextInGML("gml_Object_obj_joypadmenu_Draw_0", "joystick_buttons(obj_time.j_ch)", "11 + 1");
+ReplaceTextInGML("gml_Object_obj_joypadmenu_Draw_0", "joystick_check_button(obj_time.j_ch, i)", "gamepad_button_check(obj_time.j_ch - 1, gp_face1 + i)");
+for (var i = 0; i < 3; i++)
+{
+    ReplaceTextInGML("gml_Object_obj_joypadmenu_Draw_0", $"global.button{i} = i", $"global.button{i} = gp_face1 + i");
+    ReplaceTextInGML("gml_Object_obj_joypadmenu_Draw_0", $"string_hash_to_newline(global.button{i})", $"string(global.button{i} - gp_face1)");
+}
+
+// Use Xbox or PS4 button sprites
+if (isXbox)
+    ReplaceTextInGML("gml_Script_scr_getbuttonsprite", "os_type == os_xboxone", "true");
+else
+    ReplaceTextInGML("gml_Script_scr_getbuttonsprite", "os_type == os_ps4", "true");
+
+// Allow gamepad input for left/right heart halfs
+foreach (char half in new[] {'l', 'r'})
+{
+    foreach (char side in new[] {'u', 'd', 'l', 'r'})
+    {
+        ReplaceTextInGML($"gml_Script_scr_heart{half}_hold{side}", "global.osflavor <= 2 && ", "");
+        ReplaceTextInGML($"gml_Script_scr_heart{half}_hold{side}", "global.osflavor >= 4 && ", "");
+    }
+}
+
+if (ScriptQuestion("Enable the Dog Shrine?"))
+{
+    if (isXbox)
+    {
+        // This enables the Dog Shrine's entrance.
+        ReplaceTextInGML("gml_Object_obj_kitchenchecker_Create_0", "global.osflavor == 4 || global.osflavor == 5 || global.osflavor == 6", "true");
+        ReplaceTextInGML("gml_Object_obj_kitchenchecker_Alarm_2", "(global.osflavor == 4 || global.osflavor == 5 || global.osflavor == 6) && ", "");
+        // if in NX version, the door will get you into the ruined dog shrine
+        ReplaceTextInGML("gml_Object_obj_doorXmusicfade_Alarm_2", "if (global.osflavor == 6)", "else");
+
+        // Enable donation box trash
+        ReplaceTextInGML("gml_Object_obj_npc_room_Create_0", "global.osflavor != 4 && global.osflavor != 6", "false");
+    }
+    else
+    {
+        // This enables the Dog Shrine's entrance.
+        ReplaceTextInGML("gml_Object_obj_kitchenchecker_Create_0", "global.osflavor == 4 || global.osflavor == 5", "true");
+        ReplaceTextInGML("gml_Object_obj_kitchenchecker_Alarm_2", "(global.osflavor == 4 || global.osflavor == 5) && ", "");
+
+        // Enable donation box trash
+        ReplaceTextInGML("gml_Object_obj_npc_room_Create_0", "(global.osflavor != 4 && global.osflavor != 5) || ", "");
+    }
+}
+
+// Done.
+ScriptMessage("""
+NXTALE Enabler by Kneesnap
+Xbox and gamepad fixes by Dobby233Liu
+
+NOTE: You're not done yet!
```

Comment:
Is there a reason why we dont dynamically show the "for the switch version" / "for the xbox version" based on what version the script is currently running on?